File: pueblo-golf-tournament-api/Dto/PaymentDto.cs
using pueblo_golf_tournament_api.Base;

namespace pueblo_golf_tournament_api.Dto
{
    public class PaymentDto : BaseDto
    {
        public string? PaymentMethod { get; set; }
        public long PaymentChannelAccountId { get; set; }
        public string? PaymentReferrencePhoto { get; set; }
        public string? AccountNumber { get; set; }
        public string? AccountName { get; set; }
        public string? ReferrenceId { get; set; }
        public DateTime PaymentDate { get; set; }
    }

    public class PaymentRawDto  {
        public string? PaymentMethod { get; set; }
        public long PaymentChannelId { get; set; }
        public long PaymentChannelAccountId { get; set; }
        public string? AccountNumber { get; set; }
        public string? AccountName { get; set; }
        public string? ReferrenceId { get; set; }
        public DateTime PaymentDate { get; set; }
    }
}

File: pueblo_golf_tournament_mobile/lib/dto/model/payment-dto.dart
import 'package:pueblo_golf_tournament_mobile/base/base-dto.dart';

import 'package:json_annotation/json_annotation.dart';

part 'payment-dto.g.dart';

@JsonSerializable()
class PaymentDto extends BaseDto {
  final String paymentMethod;
  final String paymentReferrencePhoto;
  final int? paymentChannelAccountId;
  final String? accountNumber;
  final String? accountName;
  final String referrenceId;
  final DateTime paymentDate;
  PaymentDto({
    required super.id,
    required this.paymentMethod,
    required this.paymentChannelAccountId,
    required this.paymentReferrencePhoto,
    required this.referrenceId,
    required this.paymentDate,
    required this.accountNumber,
    required this.accountName,
  });

  factory PaymentDto.fromJson(Map<String, dynamic> json) =>
      _$PaymentDtoFromJson(json);
  Map<String, dynamic> toJson() => _$PaymentDtoToJson(this);
}
